homeassistant/components/mill/climate.py

```diff
@@ -195,16 +210,27 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         await self.coordinator.mill_data_connection.set_target_temperature(
             float(temperature)
         )
+        if hvac_mode := kwargs.get(ATTR_HVAC_MODE):
```

Comment:
This code is repeated in lines 124-128. Other methods below are repeated as well. While not specifically related to the reported issue, you could consider some improvements in the future. For instance, maybe LocalMillHeater could inherit from MillBaseEntity or maybe they could share some parts.

Reply:
I'm trying to keep the code footprint small, and while there's already duplication in here, I think adding a bit more isn't too bad, if it can be refactored in the future, else I risk the chance of getting too distracted from fixing this.